src/hashes/interface.ts

```diff
@@ -67,6 +68,76 @@ export interface MultihashHasher<Code extends number = number> {
  * `SyncMultihashHasher` is useful in certain APIs where async hashing would be
  * impractical e.g. implementation of Hash Array Mapped Trie (HAMT).
  */
-export interface SyncMultihashHasher<Code extends number = number> extends MultihashHasher<Code> {
+export interface SyncMultihashHasher<Code extends MulticodecCode = MulticodecCode> extends MultihashHasher<Code> {
   digest: (input: Uint8Array) => MultihashDigest<Code>
 }
+
+/**
+ * Incremental variant of the `MultihashHasher` that can be used to compute
+ * digest of the payloads that would be impractical or impossible to load all
+ * into a memory.
+ */
+export interface IncrementalMultihashHasher<
+  Code extends MulticodecCode,
+  Size extends number,
+  Digest = MultihashDigest<Code, Size>
+> {
+  /**
+   * Size of the digest this hasher produces.
+   */
+  size: Size
+
+  /**
+   * Code of the multihash
+   */
+  code: Code
+
+  /**
+   * Name of the multihash
+   */
+  name: string
+
+  /**
+   * Number of bytes that were consumed.
+   */
+  count(): bigint
+
+  /**
+   * Returns multihash digest of the bytes written so far. Should not have
+   * side-effects, meaning you should be able to write some more bytes and
+   * call `digest` again to get the digest for all the bytes written from
+   * creation (or from reset)
+   */
+  digest(): Digest
+
+  /**
+   * Encodes multihash of the bytes written so far (since creation or
+   * reset) into provided `target` at given `offset`. If `offset` not
+   * provided it is implicitly `0`.
+   *
+   * @param [offset=0] - Byte offset in the `target`.
+   */
+  readDigest(target: Uint8Array, offset?: number): this
+
+  /**
+   * Encodes raw digest (without multihash header) of the bytes written
+   * so far (since creation or reset) into provided `target` at given
+   * `offset`. If `offset` not provided it is implicitly `0`.
+   *
+   * @param [offset=0] - Byte offset in the `target`.
+   */
+  read(target: Uint8Array, offset?: number): this
+
+  /**
+   * Writes bytes to be digested.
+   */
+  write(bytes: Uint8Array): this
```

Comment:
Typically in streaming hashers this is called `update`.

Reply:
I'm fine with calling it update although I do find that name confusing personally as I think of `update` as `overwrite` as opposed to `append`.

Reply:
Right, but streaming hashers aren't appending to a buffer they are updating their internal state with the new data you pass.